packages/gtmapi/lmsrvcore/tests/test_chunk_upload.py

```diff
@@ -39,6 +20,24 @@ def test_get_temp_filename(self):
         mut = MyMutation()
         assert mut.get_temp_filename("asdf", "1234.txt") == "/tmp/asdf-1234.txt"
 
+    def test_get_safe_filename(self):
+        """Test getting safe filenames"""
+        mut = MyMutation()
+        assert mut.py_secure_filename("\n\ttest.txt") == "test.txt"
+        assert mut.py_secure_filename(u'\x00' + f"test.txt") == "test.txt"
+
+        assert mut.py_secure_filename("../../../test.txt") == "test.txt"
+        assert mut.py_secure_filename("./test.txt") == "test.txt"
+        assert mut.py_secure_filename("./../test.txt") == "test.txt"
+        assert mut.py_secure_filename("../test.txt") == "test.txt"
+        assert mut.py_secure_filename("/../test.txt") == "test.txt"
```

Comment:
These would test my above comment about `..` in the middle of paths, in case that wasn't clear.

```suggestion
        assert mut.py_secure_filename("/../test.txt") == "test.txt"
        assert mut.py_secure_filename("some/../test.txt") == "test.txt"
        assert mut.py_secure_filename("some/../dir/test.txt") == "dir/test.txt"
```

Reply:
see above, but don't think these tests need to be added because it's not the expected behavior (keeping the simpler implementation that just removes `..` and `.` from the path instead of truncating the path.